druid/src/test/java/org/apache/calcite/test/DruidAdapterIT.java

```diff
@@ -2156,6 +2156,317 @@ public RelNode apply(RelBuilder b) {
+            + "'name':'EXPR$1','fieldName':'store_cost'}],"
+            + "'intervals':['1900-01-09T00:00:00.000/2992-01-10T00:00:00.000'],"
+            + "'context':{'skipEmptyBuckets':true}}";
+
+    sql(sql).queryContains(druidChecker(expectedQuery));
+  }
+
+  /**
+   * Test to ensure that multiple aggregates with filter clauses have their filters extracted to
+   * the outer filter field for data pruning
+   * */
+  @Test public void testFilterClausesFactoredForPruning1() {
+    String sql = "select "
+            + "sum(\"store_sales\") filter (where \"store_state\" = 'CA'), "
+            + "sum(\"store_sales\") filter (where \"store_state\" = 'WA') "
+            + "from \"foodmart\"";
+    String expectedQuery = "{'queryType':'timeseries','dataSource':'foodmart','descending':false,"
+            + "'granularity':'all','filter':{'type':'or','fields':[{'type':'selector','dimension':"
+            + "'store_state','value':'CA'},{'type':'selector','dimension':'store_state',"
+            + "'value':'WA'}]},'aggregations':[{'type':'filtered','filter':{'type':'selector',"
+            + "'dimension':'store_state','value':'CA'},'aggregator':{'type':'doubleSum','name':"
+            + "'EXPR$0','fieldName':'store_sales'}},{'type':'filtered','filter':{'type':'selector',"
+            + "'dimension':'store_state','value':'WA'},'aggregator':{'type':'doubleSum','name':"
+            + "'EXPR$1','fieldName':'store_sales'}}],'intervals':"
+            + "['1900-01-09T00:00:00.000/2992-01-10T00:00:00.000'],"
+            + "'context':{'skipEmptyBuckets':true}}";
+
+    sql(sql)
+        .queryContains(druidChecker(expectedQuery))
+        .returnsUnordered("EXPR$0=159167.840144217; EXPR$1=263793.2202244997");
+  }
+
+  /**
+   * Test to ensure that multiple aggregates with filter clauses have their filters extracted to
+   * the outer filter field for data pruning in the presence of an outer filter
+   * */
+  @Test public void testFilterClausesFactoredForPruning2() {
+    String sql = "select "
+            + "sum(\"store_sales\") filter (where \"store_state\" = 'CA'), "
+            + "sum(\"store_sales\") filter (where \"store_state\" = 'WA') "
+            + "from \"foodmart\" where \"brand_name\" = 'Super'";
+    String expectedQuery = "{'queryType':'timeseries','dataSource':'foodmart','descending':false,"
+            + "'granularity':'all','filter':{'type':'and','fields':[{'type':'or','fields':[{'type':"
+            + "'selector','dimension':'store_state','value':'CA'},{'type':'selector','dimension':"
+            + "'store_state','value':'WA'}]},{'type':'selector','dimension':'brand_name','value':"
+            + "'Super'}]},'aggregations':[{'type':'filtered','filter':{'type':'selector',"
+            + "'dimension':'store_state','value':'CA'},'aggregator':{'type':'doubleSum','name':"
+            + "'EXPR$0','fieldName':'store_sales'}},{'type':'filtered','filter':{'type':'selector',"
```

Comment:
can we add at least one test where we have a nested filter ? like where store = ca or store = il